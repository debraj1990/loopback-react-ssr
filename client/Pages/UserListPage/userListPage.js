import React, {Component } from 'react'


class AdminComponent extends Component {
    render() {
        return (
            <div>
                <h1>Hi am User List Page!</h1>
            </div>
        )
    }
}

export default AdminComponent