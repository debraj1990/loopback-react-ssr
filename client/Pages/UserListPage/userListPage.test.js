import React from 'react';
import {configure, mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import { UsersList } from './userListPage';


configure({adapter: new Adapter });


describe('<UserList />', () => {
    
    let component;

    beforeEach(() => {
        component = mount(<UsersList fetchUsers={() => {} } users={[]} />)
    })

    it('should render users length   of 0 if no users is available', () => {
        expect(component.props().users.length).toBe(0)
    })

    it('should render users length of 2', () => {
        component.setProps({users: [{name:'Josh', id:2},{name:'Kay', id:5}]})
        expect(component.props().users.length).toBe(2)
    })

    it('should render when all props are passed', () => {
        const tree = toJson(component)
        expect(tree).toMatchSnapshot()
    })

})