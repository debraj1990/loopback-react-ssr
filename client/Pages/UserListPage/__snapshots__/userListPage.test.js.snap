// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserList /> should render when all props are passed 1`] = `
<UsersList
  fetchUsers={[Function]}
  users={Array []}
>
  <div>
    <HelmetWrapper
      defer={true}
      encodeSpecialCharacters={true}
    >
      <SideEffect(NullComponent)
        defer={true}
        encodeSpecialCharacters={true}
        meta={
          Array [
            Object {
              "content": "Users App",
              "property": "og:title",
            },
          ]
        }
        title="0 Users Loaded"
        titleAttributes={Object {}}
      >
        <NullComponent
          defer={true}
          encodeSpecialCharacters={true}
          meta={
            Array [
              Object {
                "content": "Users App",
                "property": "og:title",
              },
            ]
          }
          title="0 Users Loaded"
          titleAttributes={Object {}}
        />
      </SideEffect(NullComponent)>
    </HelmetWrapper>
    Here's a big list of users:
    <ul />
  </div>
</UsersList>
`;
